examples/huggingface/pytorch/text-generation/quantization/run_generation_gpu_woq.py

```diff
@@ -220,12 +220,12 @@
                 output = user_model.generate(
                     input_ids, max_new_tokens=int(args.max_new_tokens), **generate_kwargs
                 )
+                if args.device == "xpu":
+                    torch.xpu.synchronize()
                 toc = time.time()
                 gen_ids = output[0] if args.profile_token_latency else output
                 gen_text = tokenizer.batch_decode(
                     gen_ids, skip_special_tokens=True)
-                if args.device == "xpu":

```

Comment:
from doc, this API is used to wait for all kernels in all streams to be completed. so why only the line 223 needs to do synchronization but here is no need to do?

Reply:
```tic``` and ```toc``` only wraps line223, and sync must happen before ```toc``